search/pre_ranker.cpp

```diff
@@ -45,6 +45,40 @@ struct ComparePreResult1
     return linfo.m_startToken < rinfo.m_startToken;
   }
 };
+
+// Selects a fair random subset of size min(|n|, |k|) from [0, 1, 2, ..., n - 1].
+vector<size_t> RandomSample(size_t n, size_t k, minstd_rand & rng)
+{
+  vector<size_t> result(std::min(k, n));
+  iota(result.begin(), result.end(), 0);
+
+  for (size_t i = k; i < n; ++i)
+  {
+    size_t const j = rng() % (i + 1);
+    if (j < k)
+      result[j] = i;
+  }
+
+  return result;
+}
+
+void SweepNearbyResults(double eps, vector<PreResult1> & results)
```

Comment:
Results of PreRanker are PreResults, results of Ranker are Results. Sorry, but I didn't get what's wrong here.